### DISCUSSION TITLE
Why call learn_one after predict_one?

### DISCUSSION THREAD

**Discussion by cdeterman** [online-ml/river#1223] [General]
Looking through more of the API documentation I am confused by [`pure_inference_mode`](https://riverml.xyz/0.15.0/api/compose/pure-inference-mode).  It appears to suggest the `learn_one` is called whenever `predict_one` is called in the default setup.  This seems strange to me as all the examples such as the Bike Sharing example [here](https://github.com/online-ml/river/blob/main/docs/examples/bike-sharing-forecasting.ipynb) the `predict_one` is always followed by `learn_one`.  

Am I missing something here?  Does the documentation for the `pure_inference_mode` suggest that `learn_one` isn't necessary in the default mode?

**Comment by MaxHalford** [comment]
It's a bit more complicated than that. This only applies to pipelines. In a pipeline, the unsupervised parts of the pipeline (say, the scaler before the linear model) are updated when `predict_one` is called. You don't always want that to happen. For instance, you might just want to make a prediction, without updating the scaler. This is what `pure_inference_mode` enables.

Does that help? Basically, I think the confusion you have is that you expect `learn_one` to call `learn_one` on each part of the pipeline, and likewise for `predict_one`. But it's not exactly what happens under the hood.
